src/main/java/org/yearup/controllers/CategoriesController.java

```diff
@@ -58,12 +60,18 @@ public List<Product> getProductsById(@PathVariable int categoryId)
         return null;
     }
 
-    // add annotation to call this method for a POST action
-    // add annotation to ensure that only an ADMIN can call this function
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     public Category addCategory(@RequestBody Category category)
     {
-        // insert the category
-        return null;
+        try
+        {
+            return categoryDao.create(category);
```

Comment:
do we want to have this return a response entity?

Reply:
I went ahead and updated the code to return a response entity.